SplunkRumWorkspace/SplunkRum/SplunkRum/NetworkInstrumentation.swift

```diff
@@ -121,115 +121,109 @@ func startHttpSpan(request: URLRequest?) -> Span? {
     return span
 }
 
-class SessionTaskObserver: NSObject {
+class SessionTask: NSObject {
     var span: Span?
     // Observers aren't kept alive by observing...
-    var extraRefToSelf: SessionTaskObserver?
+    var extraRefToSelf: SessionTask?
     var lock: NSLock = NSLock()
     override init() {
         super.init()
         extraRefToSelf = self
     }
 
-    override func observeValue(forKeyPath keyPath: String?, of object: Any?, change: [NSKeyValueChangeKey: Any]?, context: UnsafeMutableRawPointer?) {
-        lock.lock()
-        defer {
-            lock.unlock()
-        }
-        let task = object as? URLSessionTask
-        if task == nil {
-            return
-        }
-        if span == nil {
-            span = startHttpSpan(request: task!.originalRequest)
-        }
-        // FIXME possibly also allow .canceling to close the span?
-        if task!.state == .completed && extraRefToSelf != nil {
-            endHttpSpan(span: span,
-                        task: task!)
-            task!.removeObserver(self, forKeyPath: "state")
-            extraRefToSelf = nil
+    func networkSpan(task: URLSessionTask) {
+        DispatchQueue.main.async {
```

Comment:
Is the async queue needed here?

Reply:
In order to create a span, async is required.

Reply:
Why?